docs/core/extensions/logger-message-generator.md

```diff
@@ -0,0 +1,405 @@
+
+Consider the example logging output when using the `SimpleConsole` formatter.
+
+```console
+trce: LoggingExample[9]
+      Fixed message
+warn: LoggingExample[10]
+      Welcome to Vancouver BC!
+info: LoggingExample[10]
+      Welcome to Vancouver BC!
+crit: LoggingExample[20]
+      Value is 1.234568E+004
+```
+
+Consider the example logging output when using the `JsonConsole` formatter.
+
+```json
+{
+  "EventId": 9,
+  "LogLevel": "Trace",
+  "Category": "LoggingExample",
+  "Message": "Fixed message",
+  "State": {
+    "Message": "Fixed message",
+    "{OriginalFormat}": "Fixed message"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Warning",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Information",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 20,
+  "LogLevel": "Critical",
+  "Category": "LoggingExample",
+  "Message": "Value is 1.234568E+004",
+  "State": {
+    "Message": "Value is 1.234568E+004",
+    "value": 12345.6789,
+    "{OriginalFormat}": "Value is {value:E}"
+  }
+}
+```
+
+## Summary
+
+With the advent of C# source generators, writing highly-performant logging APIs is much easier. Using the source generator approach has several key benefits:
+
+- Allows the logging structure to be preserved and enables the exact format syntax required by [Message Templates](https://messagetemplates.org).
+- Allows supplying alternative names for the template placeholders and using format specifiers.
+- Allows the passing of all original data as-is, without any complication around how it's stored prior to something being done with it (other than creating a `string`).
+- Provides logging-specific diagnostics, emits warnings for duplicate event ids.
+
+Additionally, there are benefits over manually using <xref:Microsoft.Extensions.Logging.LoggerMessage.Define%2A?displayProperty=nameWithType>:
+
+- Shorter and simpler syntax, declarative attribute usage rather than coding boilerplate.
+- Guided developer experience, the generator gives warnings to help developers do the right thing.
+- Support for an arbitrary number of logging parameters. `LoggerMessage.Define` supports a max of six.
+- Support for dynamic log level, this is not possible with `LoggerMessage.Define` alone.
+
+### Known limitations
+
+There are a few known limitations with the underlying implementation details. The first version of the generator uses `LoggerMessage.Define` where possible. The following issues are tracking potential improvements to the `Define` APIs, which could allow for better support of the source generator.
```

Comment:
We aren't supposed to discuss potential future functionality in docs.
See https://review.docs.microsoft.com/en-us/help/contribute/contribute-get-started-channel-guidance?branch=master#unauthorized-content.